aw-datastore/src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![feature(option_result_contains)]
```

Comment:
Please don't add any more nightly features, our aim is for aw-server-rust to be able to run on stable.

Reply:
I'm not 100% sure what you are referring too, using XXX.`contains` afaik, is what caused the lib.rs to be updated to add `#![feature(option_result_contains)]`.   And not sure how that stops this from being run on "Stable" -- I've been running this version of the server since I pushed my PR...

Is there a better way to see if a string contains another string that won't trigger rust to auto-add that `[feature]` flag to the lib.rs file if this is an actual issue that I'm not understanding?

Reply:
With stable I mean "rust stable", the toolchain version. We currently require rust nightly because we have one dependency which requires it (which we have plans to get rid of).

I'm not sure what you mean that rust "auto-adds" this to lib.rs, rust itself can't change the source so either you have added it youself or you have a really invasive IDE or something.

I have not used contains before, but it seems to just be syntatic sugar. You just need to add another "match" to unpack the option/result, and if it was some/ok you can just do "x == y" as usual.

> Is there a better way to see if a string contains another string

That's what String::contains does, this feature is in regards to Result::contains or Option::contains. So it's possible you have a bug here in case you are not using String::contains, since the other two checks for full equality rather than for a "substring".

Reply:
I'm guessing this was added by my IDE then.     When I have time, I'll try and remove the feature flag and see what happens, maybe the IDE guessed wrong...  :grinning: 

---

I'm not following you completely on the potential `.contains` replacement.    This isn't a `==` situation.     

The `.contains()` is the code here:
`if app.contains(&key) {` and here ` if title.contains(&key) {`  
Basically it is equivalent to Javascript `title.indexOf(key) !== -1`  where we are seeing if this specific key value is part of the app name or title.      This way if the key says "BROWSER" then if you are running an app called "INTERNET BROWSER" or "BROWSER OPERA" or "THE BROWSER IS AWESOME" all three would be ignored because it found that the string had `BROWSER` in it somewhere...    It is does string contain another string.    I'm not a rust expert by any means, but if you have a better way to see if `string A` contains `string B` then I'm willing to switch to it...  :grinning:

Reply:
I think it should be possible to continue using contains as-is in the code here and just remove the nightly flag.